activities/vidispine/client.go

```diff
@@ -13,3 +18,33 @@ func GetClient() *vidispine.VidispineService {
 
 	return vidispine.NewVidispineService(vsapiClient)
 }
+
+type WaitForJobCompletionParams struct {
+	JobID string
+}
+
+func WaitForJobCompletion(ctx context.Context, params WaitForJobCompletionParams) error {
+	logger := activity.GetLogger(ctx)
+	logger.Info("Starting WaitForJobCompletionActivity")
+
+	vsClient := GetClient()
+
+	for {
+		job, err := vsClient.GetJob(params.JobID)
+		if err != nil {
+			return err
+		}
+		if job.Status == "FINISHED" {
+			return nil
+		}
+		if job.Status != "STARTED" && job.Status != "READY" && job.Status != "WAITING" {
+			spew.Dump(job)
+			return fmt.Errorf("job failed with status: %s", job.Status)
+		}
+		activity.RecordHeartbeat(ctx, job)
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		time.Sleep(time.Second * 30)
```

Comment:
Is this correct or should the sleep be in temporal, so the job is not running while it is sleeping? Maybe we also need a "basic worker" that can do this kinds of non-process intensive stuff in parallel?

Reply:
Most of what the worker does is basic, but yes temporal sleep is probably better.